src/flowcontrol/include/rs/flowcontrol/RSPipelineManager.h

```diff
@@ -63,10 +84,13 @@ class RSPipelineManager
   // Get the index of the given annotator in this->flow_constraint_nodes.
   int getIndexOfAnnotator(std::string annotator_name);
 
+  void getCurrentAnnotatorFlow(std::vector<std::string> &annotators);
+
+public:
   // private:
   /* data */
   uima::AnalysisEngine *engine;
-  uima::internal::AggregateEngine *aengine;
+  RSParallelAnalysisEngine *aengine;
```

Comment:
why do we have two engines here?; Couldn't we do this with a single one?

Reply:
It's just the pointer of child class and parent class pointed to the same object. I respect the old implementation of RSPipelineManager, it is from the original file. The reason is that you dont have to cast up or down everytime needed